tests/aws/services/events/test_events_patterns.py

```diff
@@ -212,6 +213,39 @@ def test_invalid_json_event_pattern(self, aws_client, pattern, snapshot):
             )
         snapshot.match("invalid-pattern", e.value.response)
 
+    @markers.aws.validated
+    @pytest.mark.skipif(
+        is_old_provider(),
+        reason="V1 provider does not properly validate",
+    )
+    def test_plain_string_payload(self, aws_client, snapshot):
+        event = "plain string"
+        pattern = {"body": {"test2": [{"numeric": [">", 100]}]}}
+
+        with pytest.raises(ClientError) as e:
+            aws_client.events.test_event_pattern(
+                Event=event,
+                EventPattern=json.dumps(pattern),
+            )
+        snapshot.match("plain-string-payload-exc", e.value.response)
+
+    @markers.aws.validated
+    @pytest.mark.skipif(
+        is_old_provider(),
+        reason="V1 provider does not properly validate",
+    )
+    def test_invalid_event_payload(self, aws_client, snapshot):
+        # following fields are mandatory: `id`, `account`, `source`, `time`, `region`, `resources`, `detail-type`
```

Comment:
note: will update the comment after the review, as `resources` is not mandatory unlike what the documentation says